calm/draft/2024-04/meta/core.json

```diff
@@ -46,6 +46,9 @@
           "items": {
             "$ref": "interface.json#/defs/interface-type"
           }
+        },
+        "detailed-architecture": {
+          "$ref": "#"
```

Comment:
Do we want to use a $ref in this case? What we trying to achieve is to reduce duplication by being able to reference pre-existing CALM models. 

Whilst I can see it would be nice to $ref those documents, I'm not sure it's actually needed, we simply need a place to reference the URI of the models.

This actually ties in somewhat with having a tool that makes it easier to edit the schemas locally by being able to load all schemas from a public or private model repository (so that the URI itself doesn't need to be physically addressable).

Reply:
@rocketstack-matt my understanding of a `ref` at this level means that the `detailed-architecture` must be defined in terms of CALM - i.e. a self referential to this meta-schema.

In a pattern instantiation they could:

1) Reference a file - using a second $ref to point to their file
2) Inline define an architecture directly within this node

Reply:
@rocketstack-matt not yet, I wanted to see if you agreed with the approach. Happy for you to try an example 😄, I don't have too much time today to try it

Reply:
It feels like we need a type that represents a calm definition?